### PR TITLE
Add doc for visiblity on ElasticSearch

### PR DESCRIPTION
**Pull request by vancexu**
Share context and details about Cadence Visibility on ElasticSearch

### CODE REVIEW

File: docs/visibility-on-elasticsearch.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+In addition, Cadence want to support users to perform flexible query with multiple filters on even custom info.  
+That's why Cadence add support for enhanced visibility features on top of ElasticSearch (Note as ES below), which includes: 
+ - new visibility APIs to List/Scan/Count workflows with SQL like query
+ - search attributes feature to support users provide custom info
+ 
+# Quick Start
+## Local Cadence Docker Setup
+1. Increase docker memory to higher 6GB. Docker -> Preference -> advanced -> memory limit
+2. Get docker compose file. Run `<code>curl -O [https://raw.githubusercontent.com/uber/cadence/master/docker/docker-compose-es.yml](https://raw.githubusercontent.com/uber/cadence/master/docker/docker-compose-es.yml)</code>`
```
Comment (dmetzgar) [inline docs/visibility-on-elasticsearch.md]: Get rid of the `<code>` tags. Also, get rid of the link. It should just be:

> Run `curl -O https://raw.githubusercontent.com/uber/cadence/master/docker/docker-compose-es.yml`

You can also place this on it's own line by creating a code section with \`\`\`. That way, it's easier to copy.

File: docs/visibility-on-elasticsearch.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+3. Start cadence docker which contains Kafka, Zookeeper and ElasticSearch. Run `<code>docker-compose -f docker-compose-es.yml up</code>`
+4. From docker output log, make sure ES and cadence started correctly. If encounter disk space not enough, try `<code>docker system prune -a --volumes</code>`
+5. Register local domain and start using it. `<code>cadence --do samples-domain d re</code>`
```
Comment (dmetzgar) [inline docs/visibility-on-elasticsearch.md]: Likewise, get rid of the `<code>` tags from these list items.

File: docs/visibility-on-elasticsearch.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+Make sure Cadence CLI version is 0.6.4+
+
+### new list API examples
+
+```
+cadence --do samples-domain wf list -q 'WorkflowType = "main.Workflow" and (WorkflowID = "1645a588-4772-4dab-b276-5f9db108b3a8" or RunID = "be66519b-5f09-40cd-b2e8-20e4106244dc")'
+cadence --do samples-domain wf list -q 'WorkflowType = "main.Workflow" StartTime > "2019-06-07T16:46:34-08:00" and CloseTime = missing'
+```
+To list only open workflows, add ``CloseTime = missing`` to the end of query.
```
Comment (dmetzgar) [inline docs/visibility-on-elasticsearch.md]: Do you want backticks around CloseTime = missing?

Comment (vancexu) [reply docs/visibility-on-elasticsearch.md]: Thanks for catching this. I removed that duplicate backticks.

**Pull request merged**